lib/backend/k8s/resources/customresource.go

```diff
@@ -347,16 +363,25 @@ func (c *customK8sResourceClient) convertResourceToKVPair(r Resource) (*model.KV
 			Namespace: r.GetObjectMeta().GetNamespace(),
 			Kind:      c.resourceKind,
 		},
-		Value:    r,
 		Revision: r.GetObjectMeta().GetResourceVersion(),
 	}
 
+	if err := GetCalicoMetadataFromK8sAnnotations(r); err != nil {
+		return kvp, err
+	}
+
+	kvp.Value = r
 	return kvp, nil
 }
 
 func (c *customK8sResourceClient) convertKVPairToResource(kvp *model.KVPair) (Resource, error) {
 	resource := kvp.Value.(Resource)
+	// TODO: Figure out if Resource version is suposed to be overwritten by K8s or not.
 	resource.GetObjectMeta().SetResourceVersion(kvp.Revision)
```

Comment:
We do need to pass down the resource version into the resource definition that we pass to k8s.  This is used to perform "atomic" operations.  The K8s response will have an updated resource version.

Reply:
The ResourceVersion should be extracted from the response and included in the KVPair that is returned.

Reply:
Sounds good. Wasn't sure if I needed to store the version in the annotations as well, but from your answer, it seems we don't need to.